src/odemis/driver/xt_client.py

```diff
@@ -2179,16 +2205,42 @@ def __init__(self, name: str, role: str, parent: model.HwComponent, rng: dict =
                                                 readonly=True)
         self._updatePosition()
 
+        # update the offset if raw stage coordinates are read
+        self._get_offset_raw_coordinate_system()
+
         # Refresh regularly the position
         self._pos_poll = util.RepeatingTimer(5, self._refreshPosition, "Stage position polling")
         self._pos_poll.start()
 
+    def _get_offset_raw_coordinate_system(self) -> None:
+        """Calculate the offset in linear stage axes x and y when the stage coordinates are read in raw coordinates such
+         that it is equal to x and y when stage coordinates are read in linked coordinates."""
+        if self.raw:
+            self.parent.set_raw_coordinate_system(self.raw)
```

Comment:
Instead of changing 3 times, start with the "non raw coordinates" system, and then switch to the raw version.